apps/api/src/app/integrations/usecases/get-novu-integration/get-novu-integration.usecase.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@nestjs/common';
+import { IntegrationEntity, MessageRepository, IntegrationRepository } from '@novu/dal';
+import { GetNovuIntegrationCommand } from './get-novu-integration.command';
+import { ChannelTypeEnum, EmailProviderIdEnum } from '@novu/shared';
+import { startOfMonth, endOfMonth } from 'date-fns';
+
+@Injectable()
+export class GetNovuIntegration {
+  constructor(private messageRepository: MessageRepository, private integrationRepository: IntegrationRepository) {}
+
+  async execute(command: GetNovuIntegrationCommand): Promise<IntegrationEntity[]> {
+    const providerId = this.getProviderId(command.channelType);
+
+    if (providerId === undefined) {
+      return [];
+    }
+
+    const activeIntegrationsCount = await this.integrationRepository.count({
+      _organizationId: command.organizationId,
+      active: true,
+      channel: command.channelType,
+      _environmentId: command.environmentId,
+    });
+
+    if (activeIntegrationsCount > 0) {
+      return [];
```

Comment:
Suggesting to throw an error here aswell to better differentiate the different states

Reply:
hm I am not sure... Because having an active integration is not an error it just that they have an active integration and the active integration will be returned from GetDecryptedIntegrations and nothing else will happen.